src/Controller/Component/ModulesComponent.php

```diff
@@ -145,10 +145,40 @@ function (&$data, $key) use ($metaModules) {
             array_diff_key($metaModules, $modules),
             $pluginModules
         );
+        $this->modules = $this->modulesByRoleConfig($this->modules);
```

Comment:
we could use `$this->modules` instead of passing it by argument

```php
$this->modulesByRoleConfig();
```
Inside `modulesByRoleConfig()` change `$this->modules` directly